controller/eks-cluster-config-handler.go

```diff
@@ -109,13 +110,16 @@ func (h *Handler) recordError(onChange func(key string, config *eksv1.EKSCluster
 		}
 		if err != nil {
 			if !strings.Contains(err.Error(), "currently has update") {
-				// the update is valid in that the controller should retry but there is
-				// no actionable resolution as far as a user is concerned. An update
-				// that has either been initiated by eks-operator or another source is
-				// already in progress. It is possible an update is not being immediately
-				// reflected in upstream cluster state. The config object will reenter the
-				// controller and then the controller will wait for the update to finish.
+				// The update is valid in that the controller should retry but there is no actionable resolution as far
+				// as a user is concerned. An update that has neither been initiated by the eks-operator or another
+				// source is in already in progress. It is possible an update is not being immediately reflected in
+				// upstream cluster state. The config object will reenter the controller and then the controller will
+				// wait for the update to finish.
 				message = err.Error()
+
+				// If we have hit a retriable error, we don't know whether that error will resolve or not. Wait in this
+				// case to avoid spamming the EKS logs and thrashing other updates.
+				time.Sleep(eksRecordErrorTimeout * time.Second)
```

Comment:
All the controllers have an `EnqueueAfter` method which allows you to reenqueue the handler after a time period:

https://github.com/rancher/eks-operator/blob/1b8dfe2ea04eee43fd362c6da0e1cef2a09cb600/controller/eks-cluster-config-handler.go#L61
https://github.com/rancher/eks-operator/blob/1b8dfe2ea04eee43fd362c6da0e1cef2a09cb600/controller/eks-cluster-config-handler.go#L256

Is there a way we could use that instead? Using a sleep here will just make the handler block waiting for the time to finish instead of returning control to the controller.

Reply:
Using `eksEnqueueAfter` still spams the logs every second. Using `Sleep` prevents the log spamming from what I can see

Reply:
> LGTM, I also wonder if there is a better approach for solving this issue. Looking forward to hearing others' feedback.

I solved this at one point in encryption key rotation by setting an annotation with the time of the last sleep, and checking if the required interval (usually 1 second shorter than what we `EnqueueAfter` with) had passed since, and if it hadn't to not enqueue the object and return. There's also some fun stuff you can do with channels and tickers, but it's sort of niche and hard to get right.